libs/libcommon/include/common/fixed_mem_eq.h

```diff
@@ -0,0 +1,125 @@
+// Copyright 2022 PingCAP, Ltd.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#pragma once
+
+#include <common/defines.h>
+
+#include <cstddef>
+#include <cstdint>
+
+namespace mem_utils
+{
+
+FLATTEN_INLINE_PURE
+constexpr inline bool memcmp_eq0(const char *, const char *)
+{
+    return true;
+}
+
+FLATTEN_INLINE_PURE
+inline bool memcmp_eq1(const char * a, const char * b)
+{
+    return a[0] == b[0];
+}
+
+FLATTEN_INLINE_PURE
+inline bool memcmp_eq2(const char * a, const char * b)
+{
+    const uint16_t ax = *reinterpret_cast<const uint16_t *>(a);
+    const uint16_t bx = *reinterpret_cast<const uint16_t *>(b);
+    return ax == bx;
+}
+
+FLATTEN_INLINE_PURE
+inline bool memcmp_eq3(const char * a, const char * b)
+{
+    return memcmp_eq2(a, b) & memcmp_eq2(a + 1, b + 1);
```

Comment:
1. why use `&` instead of `&&`
2. why not use `memcmp_eq2(a,b) && memcmp_eq1(a+2,b+2)`?

Reply:
To reduce branch. If cpu failed to predict `cmp and jne`, the pipeline will be cleared and restarted. 

----

use `&&`
```
        movzx   eax, word ptr [rdi]
        cmp     ax, word ptr [rsi]
        jne     .LBB6_1
        movzx   eax, word ptr [rdi + 1]
        cmp     ax, word ptr [rsi + 1]
        sete    al
        ret
```

----

use `&`
```
        movzx   eax, word ptr [rdi]  // (1)
        movzx   ecx, word ptr [rdi + 1]  // (2)
        xor     ax, word ptr [rsi]  // (3)
        xor     cx, word ptr [rsi + 1]  // (4)
        or      cx, ax
        sete    al
        ret
```
Instructions like (1) and (2) are independent(same as (3) and (4)), can be executed parallelly.

----